server/api/utils/singletons/k8s.py

```diff
@@ -527,58 +525,67 @@ def store_secrets(
-                    exc=mlrun.errors.err_to_str(exc),
-                )
-                raise exc
-            k8s_secret = client.V1Secret(type=type_)
-            k8s_secret.metadata = client.V1ObjectMeta(
-                name=secret_name, namespace=namespace, labels=labels
-            )
-            k8s_secret.string_data = secrets
+        for attempt_number in range(retry_on_conflict_count + 1):
             try:
-                self.v1api.create_namespaced_secret(namespace, k8s_secret)
-                return mlrun.common.schemas.SecretEventActions.created
-            except ApiException as exc:
-                if exc.status == 409 and retry_on_conflict:
+                k8s_secret = self.v1api.read_namespaced_secret(
+                    name=secret_name,
+                    namespace=namespace,
+                )
+            except NotFoundError:  # If secret doesn't exist, we'll simply create it
+                k8s_secret = client.V1Secret(
+                    type=type_,
+                    metadata=client.V1ObjectMeta(
+                        name=secret_name,
+                        namespace=namespace,
+                        labels=labels,
+                    ),
+                    string_data=secrets,
+                )
+                try:
+                    self.v1api.create_namespaced_secret(
+                        namespace=namespace,
+                        body=k8s_secret,
+                    )
+                except (
+                    ConflictError
+                ) as exc:  # There was a conflict while we tried to create the secret.
                     logger.warning(
                         "Secret was created while we tried to create it, retrying...",
-                        secret_name=secret_name,
+                        secret_name=k8s_secret.metadata.name,
                         exc=mlrun.errors.err_to_str(exc),
+                        attempt=attempt_number + 1,
+                        max_attempts=retry_on_conflict_count + 1,
                     )
-                    return self.store_secrets(
-                        secret_name,
-                        secrets,
-                        namespace,
-                        type_,
-                        labels,
-                        retry_on_conflict=False,
-                    )
+                    if attempt_number == retry_on_conflict_count:
+                        # Raise exception after all retries are exhausted
+                        raise exc
+                    else:
+                        continue
+                return mlrun.common.schemas.SecretEventActions.created
+            except ApiException as exc:
+                logger.error(
+                    "Failed to retrieve k8s secret",
+                    secret_name=secret_name,
+                    exc=mlrun.errors.err_to_str(exc),
+                )
                 raise exc
+            else:  # Secret exists and we are updating it.
```

Comment:
as a convention we don't use `else` where we don't have to. matter of preference I guess